test/Transport/SmtpTest.php

```diff
@@ -199,6 +199,40 @@ public function testReceivesMailArtifacts(): void
         $this->assertStringContainsString("\r\n\r\nThis is only a test.", $data, $data);
     }
 
+    /**
+     * Fold long lines during smtp communication in Protocol\Smtp class.
+     * Test folding of long lines following RFC 5322 section-2.2.3
+     *
+     * @see https://github.com/laminas/laminas-mail/pull/140
+     */
+    public function testLongLinesFoldingRFC5322(): void
+    {
+        $message = $this->getMessage();
+        // Create buffer of 8192 bytes (PHP_SOCK_CHUNK_SIZE)
+        $buffer = str_repeat('0123456789abcdef', 512);
+        $this->assertEquals(8192, strlen($buffer));
+
+        $headerValue = substr($buffer, 0, 1000);
+        $bufferSizeHeaderValue = substr($buffer, 0, 8192 - strlen('X-Exact-Length') - 2);
+        $message->getHeaders()->addHeaders([
+            'X-Ms-Exchange-Antispam-Messagedata' => $headerValue,
+            'X-Exact-Length' => $bufferSizeHeaderValue,
+        ]);
+
+        $this->transport->send($message);
+        $data = $this->connection->getLog();
+
+        $lines = explode("\r\n", $data);
+        foreach ($lines as $line) {
```

Comment:
In the way the test is written, somehow someday `$data` could turn empty, and the assertions inside `foreach` go silently ignored. Please add an assertion to verify there's a content, the more precise the better:

```suggestion
        $this->assertCount(123, $lines);
        foreach ($lines as $line) {
```

Reply:
@Slamdunk good catch! Committed.